### ISSUE TITLE
"Metadata index not found" with latest Mailpile version (accounts and preferences are not loaded)

### ISSUE THREAD

**Issue by Kabouik** [mailpile/Mailpile#1859]
Hi,

I was using Mailpile for the past few weeks and since I am using a laptop and have not enough disk space in `~/.local/share/Mailpile`, I decided to symlink this folder to another SSD where I have more free space. The symlink points to `/media/stockage/Data/Emails/Mailpile/DATA/Mailpile`, and `/media/stockage/Data/Emails/Mailpile/` happens to be the directory in which lied my Mailpile files cloned with git.

I understand that having my Mailpile database in the same directory as the git files is not a good idea, so when some changes were pushed in the repository last week-end, I decided to git clone Mailpile with the latest changes into another folder: `~/.mailpile`, and reinstalled Mailpile using the [Getting started](https://github.com/mailpile/Mailpile/wiki/Getting-started-on-Linux) page. I kept the symlink from `~/.local/share/Mailpile`. I did not move my `/media/stockage/Data/Emails/Mailpile/DATA/Mailpile` folder, just made a copy of it just in case.

Now when I try to run Mailpile from the new installation directory with the new _virtualenv_, I get the following message: 

> Metadata index not found: /home/kabouik/.mailpile/mailpile.idx                 
> Elapsed: 0.040s (Loaded metadata, 0 messages)                                  
> Elapsed: 0.005s (health: We are healthy!)                                      
> Metadata index not found: /home/mathieu/.mailpile/mailpile.idx

The thing is this `mailpile.idx` does exist but is located in `/media/stockage/Data/Emails/Mailpile/DATA/Mailpile/default`, but it never was in the installation folder as suggested by the error, even with the previous Mailpile version that used to work correctly. I tried copying this .idx file in `~/.mailpile`, but then the console says that the metadata is corrupted and that it'll try to cope, but didn't seem to do anything after a few minutes. 

Weirdly enough, when I try to run Mailpile from the previous version (previous git folder), I get the same error message, but I suppose it's because the _virtualenv_ is still associated with the new installation in the new folder? 

I tried restoring my `/media/stockage/Data/Emails/Mailpile/DATA/` folder from the backup in case the new Mailpile version corrupted it, but same result. I also tried changing the MAILPILE_HOME directly to my database instead of going through the symlink, using `export MAILPILE_HOME=/path/to/database`, but no success either.

What should I do to get Mailpile read my database again? At the moment, Mailpile thinks I have no profile yet and asks me to select the language, then to set a passphrase, and then offers me to add e-mail accounts. Since importation took several days, I would like to avoid starting from scratch again.

I am sure these issues are cause by mistakes I made, including putting my symlink in the git folder which was kind of dumb in the first place. However, I'd be grateful if anyone could help find solving this issue.

Cheers,

Kabouik

**Comment by Kabouik** [comment]
Turns out this was caused by the new Mailpile version being installed in `~/.mailpile`. After reading the documentation here over and over, I noticed that some earlier versions of Mailpile used this directory for configuration, and that latest versions still check whether this directory exists for retrocompatibility.

I guess seeing the directory existed but not finding the expected files therein caused the issue. I installed the latest Mailpile version in another folder and it did find my database.
